### PR TITLE
fix: lock provisioning order of user disk partitions

### LINKED ISSUE #9877: user disks multiple partitions on the same disk get mixed up

### PR DESCRIPTION
**Pull request by smira**
Fixes #9877

As a side-effect, fix alignment of user disks for newer QEMU versions.

### CODE REVIEW

File: cmd/talosctl/cmd/mgmt/cluster/create.go [hunk 1]
```diff
@@ -1211,8 +1213,9 @@ func getDisks() ([]*provision.Disk, error) {
 		}
 
 		disks = append(disks, &provision.Disk{
-			// add 1 MB to make extra room for GPT and alignment
-			Size:            diskSize + 2*1024*1024,
+			// add 2 MB per partition to make extra room for GPT and alignment
+			// align to 4MB due to QEMU limitations
+			Size:            (diskSize + 2*1024*1024*uint64(len(diskPartitions)+1) + QEMUAlignment - 1) / QEMUAlignment * QEMUAlignment,
```
Comment (dsseng) [inline cmd/talosctl/cmd/mgmt/cluster/create.go:1218]: Maybe extract this variable and create it step by step like this for clarity:
```go
const GPTExtraSpace = 2*1024*1024
size := diskSize + GPTExtraSpace*uint64(1+len(diskPartitions))
size = (size + QEMUAlignment - 1) / QEMUAlignment * QEMUAlignment
```

**Review by dsseng — APPROVED** [review]

File: cmd/talosctl/cmd/mgmt/cluster/create.go (hunk 1 above)
Comment (smira) [reply cmd/talosctl/cmd/mgmt/cluster/create.go:1218]: but I can do the change you proposed as well

Comment (dsseng) [reply cmd/talosctl/cmd/mgmt/cluster/create.go:1218]: Okay, then that'd be a different name, yet still not a number

**Review by DmitriyMV — APPROVED** [review]

File: cmd/talosctl/cmd/mgmt/cluster/create.go (hunk 1 above)
Comment (smira) [reply cmd/talosctl/cmd/mgmt/cluster/create.go]: I refactored this completely moving the QEMU alignment into QEMU provisioner applying to any disk types, while GPT gap is still here

**Pull request merged**